simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/PayloadFactory.java

```diff
@@ -117,6 +119,20 @@ TransactionEventRequest createTransactionEventEnded(Evse evse, Integer connector
         return createTransactionEvent(evse.getId(), connectorId, reason, transactionData, TransactionEventRequest.EventType.ENDED, currentDateTime, evse.getSeqNoAndIncrement());
     }
 
+    NotifyReportRequest createNotifyReportRequest(@Nullable Integer requestId, boolean tbc, int seqNo, ZonedDateTime generatedAt, List<ReportDatum> reportData) {
```

Comment:
Exactly what is happening now.

I'm using nullable everywhere I accept null (requestId) && I'm using primitives wherever possible (seqNo, tbc), since they can't be null.

Reply:
The problem with @Nullable is that we dont use those annotations in the code. The code ideally should be using the same style. If non-null parameters are used then we need to mark our arguments with @Nonnull annotation also. 

And also If they cant be null then please use requireNonNull checks where applicable

Reply:
ok. I think it would be faster for you to write the method signature you want. Keep in mind that in 

` createNotifyReportRequest(@Nullable Integer requestId, boolean tbc, int seqNo, ZonedDateTime generatedAt, List<ReportDatum> reportData)`

Everything except requestId is required.